### ISSUE TITLE
Refactor Tree.export

### ISSUE THREAD

**Issue by icristescu** [mirage/irmin#1850]
for the case where the node is of type key (see https://github.com/mirage/irmin/pull/1795#discussion_r881642946)

**Comment by Ngoguey42** [comment]
# Is that solved by #1795? I can't remember

**Comment by icristescu** [comment]
no, its a refactoring possible on top of #1795. It's not possible that `t.info.ptr` is a key but not `t.v` , so a (small?) simplification is possible in the export function.

**Comment by Ngoguey42** [comment]
I see. Thanks
